### ISSUE TITLE
packing loss 的归一化问题

### ISSUE THREAD

**Issue by Chandler-Bing** [THUDM/LongAlign#12]
[这里的loss计算](https://github.com/THUDM/LongAlign/blob/main/modeling_chatglm.py#L900)是不是应该归一化一下

`loss = (loss * shift_weights).sum()` -> `loss = (loss * shift_weights).sum()  / shift_weights.sum()`  

把loss归一化到token粒度
前一种方式，loss的scale偏大，而且反向传播梯度也会偏大。而且极限情况下，假设每个样本只有1个token，这个batch的loss会爆炸

**Comment by bys0318** [comment]
这里的`shift_weights`已经经过归一化了。每个sample的weight加起来为1。

**Comment by YueChenkkk** [comment]
我想问下packing loss的训练模式下，对应的attention_mask设置在代码中有体现吗？我在modeling_llama.py好像没有找到packing情形下设置attention_mask的代码
